crates/api/src/id.rs

```diff
@@ -0,0 +1,288 @@
+//! Types dealing with data identity or hashing.
+
+macro_rules! imp_deref {
```

Comment:
`derive_more` can handle this and the `1.0` library has been split up so that you can just import what you need

Reply:
We agreed to go without this dep for now, but we can always choose differently in the future.